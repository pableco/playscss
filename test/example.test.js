/* eslint-env node, jest */
/* eslint "import/no-extraneous-dependencies": ["error", {"devDependencies": true }] */
/* eslint-disable no-unused-expressions */
/* eslint-disable max-statements */

import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import Example from '../src/Model';

const props = {
    classname: '--news',
};

let container = null;
beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

describe('component snapshoot', () => {

    it('render without props', () => {
        act(() => {
            render(<Example />, container);
        });

        expect(
            pretty(container.innerHTML),
        ).toMatchSnapshot();
    });

    it('render with all props', () => {
        act(() => {
            render(<Example {...props}/>, container);
        });

        expect(
            pretty(container.innerHTML),
        ).toMatchSnapshot();
    });
});

describe('Example', () => {

    it('render changing cta text', () => {
        act(() => {
            render(<Example cta={'click here'} />, container);
        });
        expect(container.querySelector('button').textContent).toBe('click here');
    });

});
