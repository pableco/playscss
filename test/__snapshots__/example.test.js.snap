// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component snapshoot render with all props 1`] = `
"<div id=\\"holder\\">
  <p class=\\"message\\"><span class=\\"first\\">Ruben's</span><span class=\\"second\\">look</span></p>
</div>"
`;

exports[`component snapshoot render without props 1`] = `
"<div id=\\"holder\\">
  <p class=\\"message\\"><span class=\\"first\\">Ruben's</span><span class=\\"second\\">look</span></p>
</div>"
`;
